### PR TITLE
Openseadragon

### PR DESCRIPTION
**Pull request by gerardmrk**
Working example, no easier alternative atm other than uploading high-res image to the repo.

### CODE REVIEW

File: src/shared/views/plugins/DeepZoom/DeepZoom.js [hunk 1]
```diff
@@ -1,17 +1,43 @@
-import React from 'react'
+import React, { Component, PropTypes } from 'react'
```
Comment (ro-savage) [inline src/shared/views/plugins/DeepZoom/DeepZoom.js]: We just import `React` then use `React.Component` & `React.PropTypes`.

Its really just a personal preference, but we have standardized on that.

File: src/shared/views/plugins/DeepZoom/DeepZoom.js [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
-  return (
-    <div className={styles.holder}>
-      <iframe width="100%" height="100%" src={props.url} frameBorder="0" allowFullScreen />
-    </div>
-  )
-}
+const osdId = 'osd'
+
+class DeepZoom extends Component {
+  static propTypes = {
```
Comment (ro-savage) [inline src/shared/views/plugins/DeepZoom/DeepZoom.js]: Again, we are using `DeepZoom.propTypes = {}`.
Personal preference, but let's keep it consistent in this codebase.

File: src/shared/views/plugins/DeepZoom/DeepZoom.js [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
+    return (
+      <div
+        ref={elm => { this[osdId] = elm }}
+        className={styles.holder}
+      />
+    )
+  }
 }
 
 export default DeepZoom
```
Comment (ro-savage) [inline src/shared/views/plugins/DeepZoom/DeepZoom.js:17]: I would simplify this component.

 Maybe I am missing something and there is a part I dont understand?

Wanna chat on slack about it @gerardmrk ?

``` js
import React from 'react'
import OpenSeaDragon from 'openseadragon'

import styles from './DeepZoom.css'

const DeepZoom = (props) => {
  const mountOpenSeaDragon = (el, url) => {
    OpenSeaDragon({
      element: el,
      prefixUrl: '/static/images/dzi-controls/',
      tileSources: url,
      maxZoomPixelRatio: Infinity,
    })
  }

  return (
    <div
      ref={el => { mountOpenSeaDragon(el, props.url) }}
      className={styles.holder}
    />
  )
}

DeepZoom.propTypes = {
  url: React.PropTypes.string.isRequired,
}

export default DeepZoom
```

File: converttodzi.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable */
```
Comment (ro-savage) [inline converttodzi.js:1]: Does this file create a .dzi but searching every file/folder in `image/scollection-item` ?
Or does it actually cut up the images too?

File: src/static/images/collection-items/aircraft-carrier.dzi [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ro-savage) [inline src/static/images/collection-items/aircraft-carrier.dzi]: I don't understand how the dzi file knows where too look?

**Review by ro-savage — CHANGES_REQUESTED** [review]
Awesome work Gerard. You got it working damn quickly!

So there a few things that I dont understand and would like to know more about. And a couple suggestion on how to simplify code / come in line with our coding stytle.

A few issues too:
- [ ] `npm run ssr-start` is broken because our server doesnt understand 

``` js
 7 |
   8 | class DeepZoom extends Component {
>  9 |   static propTypes = {
     |                    ^
  10 |     url: PropTypes.string.isRequired,
  11 |   }
```
- [ ] The server side rendering also breaks because openseadragon wants to access `window` which doesn't exist.
- [ ] Has this been tested on android and iphone?
- [ ] I can't drag to move around the image when zoomed?
- [ ] Lets leave in the original deepzoom for now, because they are real deep zooms and implement this deepzoom on the maps page.

Let's see if we can get it working using cloudinary hosted files. But this can be a separate PR.

Again, awesome work Gerard!

File: converttodzi.js (hunk 4 above)
Comment (gerardmrk) [reply converttodzi.js:1]: It uses sharp to convert any compatible image files it finds in the `collection-items` directory to dzi. It generates their respective tiles too, you can find them in folders named after the original files and postfixed with `_files`. If you open the network tab in Chrome dev tools, you'll see that it successfully calls the appropriate tiles with status 200 everytime you zoom-in. I had to use a high-res image to properly show proof-of-concept; the original images (the aircraft carrier for example) weren't high-res enough that Nick and I initially thought it wasn't calling the right tiles etc.

File: src/static/images/collection-items/aircraft-carrier.dzi (hunk 5 above)
Comment (gerardmrk) [reply src/static/images/collection-items/aircraft-carrier.dzi]: the dzi file itself is actually an XML file, not an image file. Had to do bit of reading bout it too to realize that's how it's supposed to work. You can open the file in Atom (it won't crash) and actually see the XML attributes. After the client calls the `.dzi` file, it immediately looks into the same location (i.e. `/static/images/collection-items/..`) for the tiles. and renders them accordingly.

File: converttodzi.js (hunk 4 above)
Comment (gerardmrk) [reply converttodzi.js:1]: Sharp is Node.js's image-processing tool. All it's dependencies are statically-linked, so you won't have to worry bout `brew install imagemagick` or `libdv` etc, it auto-installs and keeps it in node_modules. This was just a temporary setup so we can use this to immediately convert images we want to dzi. Sharp is faster than all image-processing packages out there for node (of the few that exists) and apparently even more performant than graphicsmagick and imagemagick in certain scenarios. that's why I chose it. It's also currently the only package that has all its dependencies statically linked so completely portable.

File: webpack.universal.js [hunk 6]
```diff
@@ -84,26 +86,12 @@ const webpackconfig = {
...
-      mangle: {
-        screw_ie8: true,
-        keep_fnames: true,
-      },
-    }),
     new ExtractTextPlugin('../common.css', {
       allChunks: true,
     }),
+    new webpack.DefinePlugin({ 'process.env': {
+      IMAGE_STORAGE_URL: JSON.stringify(process.env.IMAGE_STORAGE_URL),
```
Comment (ro-savage) [inline webpack.universal.js:93]: Seems weird to pass as a var to webpack. In fact is weird. We should just a config while in the app itself.

But works for now.

File: src/shared/views/plugins/DeepZoom/osd.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
```
Comment (ro-savage) [inline src/shared/views/plugins/DeepZoom/osd.js:1]: Nice work. All working smoothly. Lets take a look at this in more detail when I am back.

Ace work getting it work. It rocks!

File: src/shared/views/map/MapTab/MapTab.js [hunk 8]
```diff
@@ -1,11 +1,81 @@
 import React from 'react'
```
Comment (ro-savage) [inline src/shared/views/map/MapTab/MapTab.js:1]: Awesome nice work again. I would say eventually we'll take this out and make it generic and then use it on the map page.

That way we get to use the controls you created everywhere.

Look and works really nicely!

**Review by ro-savage — APPROVED** [review]

**Comment by ro-savage** [comment]
Awesome work gerard!! Approved and merged.

**Pull request merged**